### DISCUSSION TITLE
Will traefik-certdumper continue to be supported?

### DISCUSSION THREAD

**Discussion by witten** [Mailu/Mailu#3161] [General]
I just switched my existing Mailu server over to use Traefik and [traefik-certdumper](https://mailu.io/2.0/reverse.html#traefik-as-reverse-proxy) as per the 2.0 docs, and so far they work great. However, I see that in Mailu master, the certdumper approach to using Traefik with Mailu is completely gone, [replaced with Traefik talking proxy protocol with Mailu's front container](https://mailu.io/master/reverse.html#traefik-as-reverse-proxy). I initially experimented with that approach and was never able to get it to work, which is why I went back to certdumper.

So my question: Will certdumper continue to be supported in Mailu 2.1+?

**Comment by nextgens** [comment]
No, there is no plan to keep support for it.

What's documented in master is the way forward but it requires running master; this is probably why you have been unable to get it to work.

**Reply by witten** [reply]
Thanks for the response. I've tried the new approach with Mailu master and no luck. I'll reevaluate though when it actually makes its way into a release. For what it's worth, the certdumper approach is working great for me now and I would love to see it continue to be supported even just as a "backup" or "you're on your own" approach.

**Reply by patzm** [reply]
yes same here. The [proposal](https://mailu.io/master/reverse.html) for the new version is to let Mailu manage certificates and basically, wildcard forward everything (`HostSNI`). Also that requires forwarding the port `80` to the container, which creates routing conflicts with Traefik-managed letsencrypt certificates (see https://github.com/Mailu/Mailu/pull/3440#discussion_r1752423094).

I get that having a fully container reverse proxy, router, and certificate management is nice to host Mailu on a server that only does mail. However, it is quite difficult to have multiple services then that all inherit the same certificates and routing infrastructure.

What would it take to bring back the old `certdumper` approach? Like @witten proposed.

**Reply by nextgens** [reply]
What was referred to as master has become 2024.06 ... we do not support certdumper anymore.

> Also that requires forwarding the port 80 to the container, which creates routing conflicts with Traefik-managed letsencrypt certificates

It does not; delegate all the Mailu FQDNs to Mailu using `Host` directives, combine them using logical operators (to accommodate the v3 syntax) and there will be no problem on port 80. The `OR Path` directive is just in case some FQDNs are forgotten and not necessary.

> it is quite difficult to have multiple services then that all inherit the same certificates and routing infrastructure

That is not a design goal here.

**Reply by patzm** [reply]
Yep I was running `2024.06`.

* Do you mean `Host` or `HostSNI`? The example doc says only the port `80` routing shall use `Host`, the others `HostSNI`. If that is so, how do I avoid a conflict with traefik's own ?
  ![image](https://github.com/user-attachments/assets/dfac0cc6-f392-4817-8874-f44de84b4fda)

* Is traefik v3 required? I was running `v2.11`. I can do that update before.
* ok fair thanks for making that clear 🙏

**Reply by nextgens** [reply]
You need ``Host`` for port 80 and ``HostSNI`` for other ports.

You do not need ``PathPrefix`` at all if you are diligent at listing the required hostnames; If you can't be I suggest configuring traefik to use a different type of challenge.

No, traefikv3 is not required, v2 should be fine.
